connection/connection.go

```diff
@@ -0,0 +1,69 @@
+package connection
+
+import (
+	"context"
+	"net"
+	"strings"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/kubernetes-csi/csi-lib-utils/protosanitizer"
+	"google.golang.org/grpc"
+)
+
+const (
+	// Interval of logging connection errors
+	connectionLoggingInterval = 10 * time.Second
+)
+
+// Connect opens insecure gRPC connection to a CSI driver. Address must have either '<protocol>://' prefix, or be
+// a path to a socket file. The function tries to connect indefinitely every second until it connects.
+// The function automatically adds interceptor for gRPC message logging.
+func Connect(address string, dialOptions ...grpc.DialOption) (*grpc.ClientConn, error) {
```

Comment:
Is it worthwhile to pass in a context? Then a caller can decide to abort after a certain time (`context.WithTimeout`) or run `Connect` in a goroutine and cancel the operation (`context.WithCancel`).

Reply:
IMO, the point of this function is to have the same behavior in all sidecar containers. Adding optional parameters works against this unification.

Reply:
We can have the same behavior in all sidecars by calling the function the same way, while still writing the underlying function in a more flexible way. But I get your point.

Let me add another reason for passing in a context: when discussing glog vs. klog, it was mentioned that it would be good to pass in a logger via the context instead of using the same global instance. If we had that now, a unit test could select a mock logger to test that the warning messages are printed.

But we don't have that now, so I'm fine with leaving out the context and adding it when really needed. We can deal with the API change in the sidecars on a case-by-case basis, should we do it later.